src/error.rs

```diff
@@ -0,0 +1,7 @@
+use std::time::Duration;
+
+#[derive(Debug, PartialEq, Eq)]
```

Comment:
Shall we add `Debug` to `RateLimiter` and `TokenBucket`? It seems to be a best practice to add for each data type you introduce. It's super weird to ask users to put it literally everywhere and do not make it a default. :-/

Reply:
Sigh... We can't simply derive `Debug` because of our `clock` attribute :( If you don't mind, let me merge this as is, and tackle that separately.

Reply:
Nope, I don't mind. I was just curious about best practices.